### ISSUE TITLE
Expo-Cli cannot find simulator

### ISSUE THREAD

**Issue by crathor** [expo/expo-cli#469]
## Description

Once starting my expo app and clicking Run on IOS Simulator I receive an error saying simulator is not installed. I have followed all instructions with the error and can confirm I have a simulator installed.

MacOS Mojave Version 10.14.3 (18D109). This issue began after the newest xcode update.

## Expected Behavior

I press on Run on IOS simulator and expect the simulator to start. It returns an error saying simulator not installed

## Observed Behavior

What actually happened when you performed the above actions?

If there's an error message, please paste the _full terminal output and error message_ in this code block:

```
Simulator not installed. Please visit https://developer.apple.com/xcode/download/ to download Xcode and the iOS simulator. If you already have the latest version of Xcode installed, you may have to run the command `sudo xcode-select -s /Applications/Xcode.app`.
```

## Environment

Please run `expo diagnostics` in the project folder and copy the output here: 
Expo CLI 2.13.0 environment info:
    System:
      OS: macOS 10.14.3
      Shell: 2.7.1 - /usr/local/bin/fish
    Binaries:
      Node: 8.14.0 - /usr/local/opt/node@8/bin/node
      Yarn: 1.13.0 - /usr/local/bin/yarn
      npm: 6.4.1 - /usr/local/opt/node@8/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.1 AI-173.4907809
      Xcode: 10.2/10E125 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^32.0.0 => 32.0.6 
      react: 16.5.0 => 16.5.0 
      react-native: https://github.com/expo/react-native/archive/sdk-32.0.1.tar.gz => 0.57.1 
      react-navigation: ^3.0.9 => 3.3.2 
    npmGlobalPackages:
      expo-cli: 2.13.0

Also specify:

- Phone/emulator/simulator platform, model and version:
Running Iphone 5s
Version 10.2 (SimulatorApp-880.5 CoreSimulator-587.35)

## Reproducible Demo

Please provide a minimized reproducible demonstration of the problem you're reporting.

Issues that come with minimal repro's are resolved much more quickly than issues where a maintainer has to reproduce themselves.

**Comment by AdamJNavarro** [comment]
Hey @crathor,

Can you try running `Erase all contents and settings` under the Hardware tab on your simulator and see if that resolves the issue?

Cheers,
Adam

**Comment by alivanov** [comment]
Have the same problem...
It says 

> Simulator is installed but is identified as 'com.apple.CoreSimulator.SimulatorTrampoline'; don't know what that is.

`Erase all contents and settings` did not help

**Comment by cruzach** [comment]
@crathor , @alivanov -

Temporary fix is to have the simulator already open when trying to "Run on iOS Simulator"

Easy way to do this is by running `open -a Simulator` before `expo start`

**Comment by crathor** [comment]
@AdamJNavarro - 
No luck on the Erase all contents and settings.

@cruzach - 
Thanks that solution was way better than mine 👍

**Comment by eitanfr** [comment]
Facing the same issue, 
Also I am having another error message on top of the one that was mentioned: 
`Simulator is installed but is identified as 'com.apple.CoreSimulator.SimulatorTrampoline'; don't know what that is.`

![image](https://user-images.githubusercontent.com/9163916/55275552-e8a4eb00-52f8-11e9-9747-7e4a6d09b823.png)

**Comment by ivivanov18** [comment]
Had the same problem. The solution from @cruzach (launch iOS simulator before `expo start`) worked for me too.

**Comment by methodbox** [comment]
For a fix until this is actually fixed in Expo, see this: https://github.com/expo/expo/issues/1203#issuecomment-478243565

**Comment by SantiMA10** [comment]
As @methodbox mentioned, rebooting solves the problem but is not a permanent solution, at least, for me 😞

**Comment by methodbox** [comment]
@SantiMA10

That’s not the entire solution. Read the link I posted. You need to run the command Expo tells you to run when you attempt to open a Simulator and it fails. *Then* reboot.

This has fixed it **permanently** for me.

**Comment by SantiMA10** [comment]
I runned the command, then rebooted and a hour or so later it failed again. So I runned again the command....

**Comment by methodbox** [comment]
@SantiMA10

Did you run this with sudo or did you switch to root first with `sudo su`?

I did not switch to root. The context can change the permissions with which it’s run; that’s the only thing I can think my be different.

**Comment by SantiMA10** [comment]
I run it with sudo.

**Comment by elioncho** [comment]
+1

**Comment by laustdeleuran** [comment]
+1

Having the same issue, and the fix outlined by @methodbox (running the cmd + reboot) only works temporarily for me. The issue keeps coming up.

```
Simulator is installed but is identified as 'com.apple.CoreSimulator.SimulatorTrampoline'; don't know what that is.
Simulator not installed. Please visit https://developer.apple.com/xcode/download/ to download Xcode and the iOS simulator. If you already have the latest version of Xcode installed, you may have to run the command `sudo xcode-select -s /Applications/Xcode.app`.
```

**Comment by ecbrodie** [comment]
I can confirm that I avoid the issue when I execute `open -a Simulator` to launch the iOS simulator first before `expo start`. This was after I ran the `sudo xcode-select ...` command, not sure if that is relevant.

**Comment by SantiMA10** [comment]
As @cruzach, @laustdeleuran and @ecbrodie say when I launch the emulator first I didn't have any problems.

Anyway, I still prefer that this work as always with `expor start` launching the emulator, hope we get a fix sometime soon.

**Comment by cruzach** [comment]
Hey everyone, 

I'm sorry about this, it's definitely an inconvenience to need to open the simulator manually each time, and we want to make the development process as smooth as possible. I'm working on a fix and hopefully will have it out soon 👍

**Comment by ecbrodie** [comment]
I have a hunch that this bug was caused by the most recent MacOS update, which was a Command Line Tools update. Maybe something got renamed and thus broke something hard coded in Expo.

**Comment by methodbox** [comment]
@ecbrodie

That is definitely the case. There were breaking changes in the latest Xcode update, which is known.

**Comment by cruzach** [comment]
After updating macOS to 10.14.4, this is problem is gone.

I believe it was an issue on Apple's end where the id of the `Simulator` app was different depending on if the Simulator application itself was running or not. Anyways, by updating your mac to the [latest release](https://developer.apple.com/documentation/macos_release_notes/macos_mojave_10_14_4_release_notes), you should be good to go.

This may also solve this [issue](https://github.com/facebook/react-native/issues/23878) filed at the React Native repo

**Comment by ecbrodie** [comment]
@cruzach I'm still running into the issue. I already had the latest version of _everything_ installed when I reported it yesterday and nothing new has been released since then.

* MacOS 10.14.4
* Xcode 10.2
* iOS 12.1 Simulator on Xcode
* expo-cli 2.14.0
* yarn 1.15.2

This is still an issue, please reopen.

**Comment by SantiMA10** [comment]
+1 @ecbrodie

**Comment by cruzach** [comment]
@ecbrodie & @SantiMA10 ,

Could you run `expo diagnostics` and paste the output here? 

Thanks!

**Comment by youfoundron** [comment]
Seeing this issue as well

**Comment by ecbrodie** [comment]
@cruzach results here:

```
$ expo diagnostics

  Expo CLI 2.14.0 environment info:
    System:
      OS: macOS 10.14.4
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 10.14.2 - ~/.nvm/versions/node/v10.14.2/bin/node
      Yarn: 1.15.2 - /usr/local/bin/yarn
      npm: 6.4.1 - ~/.nvm/versions/node/v10.14.2/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.3 AI-182.5107.16.33.5314842
      Xcode: 10.2/10E125 - /usr/bin/xcodebuild
    npmPackages:
      @types/expo: ^31.0.7 => 31.0.12
      @types/react: ^16.7.18 => 16.8.3
      expo: ^32.0.0 => 32.0.6
      react: 16.5.0 => 16.5.0
      react-native: https://github.com/expo/react-native/archive/sdk-32.0.0.tar.gz => 0.57.1
```

**Comment by jordymeow** [comment]
+1

**Comment by SantiMA10** [comment]
Sorry, but I was OOO until now.

```
Expo CLI 2.14.0 environment info:
    System:
      OS: macOS 10.14.4
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 10.15.0 - ~/.nvm/versions/node/v10.15.0/bin/node
      Yarn: 1.13.0 - /usr/local/bin/yarn
      npm: 6.4.1 - ~/.nvm/versions/node/v10.15.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.3 AI-182.5107.16.33.5264788
      Xcode: 10.2/10E125 - /usr/bin/xcodebuild
    npmPackages:
      @types/expo: ^32.0.0 => 32.0.0 
      @types/react: ^16.7.0 => 16.8.2 
      @types/react-native: ^0.57.34 => 0.57.34 
      @types/react-navigation: ^3.0.1 => 3.0.1 
      expo: ^32.0.0 => 32.0.4 
      react: 16.5.0 => 16.5.0 
      react-native: https://github.com/expo/react-native/archive/sdk-32.0.0.tar.gz => 0.57.1 
      react-navigation: ^3.3.2 => 3.3.2 
    npmGlobalPackages:
      expo-cli: 2.14.0
`
``

**Comment by methodbox** [comment]
As per @cruzach - https://github.com/expo/expo-cli/pull/489

I’m going to try just setting a environment variable alias for this; seems like this would achieve the same thing.

**Comment by igorclark** [comment]
Seeing this too after Xcode 10.2 update.

```
  Expo CLI 2.14.0 environment info:
    System:
      OS: macOS 10.14.4
      Shell: 5.0.3 - /usr/local/bin/bash
    Binaries:
      Node: 11.12.0 - /usr/local/bin/node
      Yarn: 1.15.2 - /usr/local/bin/yarn
      npm: 6.7.0 - /usr/local/bin/npm
    IDEs:
      Xcode: 10.2/10E125 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^32.0.0 => 32.0.6
      react: 16.5.0 => 16.5.0
      react-native: https://github.com/expo/react-native/archive/sdk-32.0.0.tar.gz => 0.57.1
```

**Comment by methodbox** [comment]
Since this issue is CLOSED and pending merge on #489, you may just want to watch that issue rather than waste any time reporting system info here.

Once that’s merged it should be fixed. I will be trying the environment variable later tonight as a work-around (way to busy with actual coding ATM) and will post back on results.

https://github.com/expo/expo-cli/pull/489
